packages/mimir/src/rules/no-useless-spread.ts

```diff
@@ -15,20 +24,58 @@ export class Rule extends AbstractRule {
                 match.index,
             )!;
             if (
-                (
-                    isSpreadElement(node) && node.expression.kind === ts.SyntaxKind.ArrayLiteralExpression ||
-                    isSpreadAssignment(node) && node.expression.kind === ts.SyntaxKind.ObjectLiteralExpression
-                ) &&
+                ((isSpreadElement(node) &&
+                    isArrayLiteralExpression(node.expression) &&
+                    (!node.expression.elements.some(isOmittedExpression) || isReassignmentTarget(node))) ||
+                    (isSpreadAssignment(node) && isObjectLiteralExpression(node.expression))) &&
                 node.expression.pos - 3 === match.index
             )
                 this.addFailureAtNode(node, FAILURE_STRING, removeUselessSpread(node));
         }
     }
 }
 
-function removeUselessSpread(node: ts.SpreadElement | ts.SpreadAssignment): Replacement[] {
-    return [
+function removeUselessSpread(node: ts.SpreadElement | ts.SpreadAssignment): Replacement | Replacement[] {
+    /* Handles edge cases of spread on empty array literals */
+    if (isSpreadElement(node) && isArrayLiteralExpression(node.expression) && node.expression.elements.length === 0)
+        return removeUselessSpreadOfEmptyArray(node);
+
+    const replacements = [
         Replacement.delete(node.getStart(), node.expression.getStart() + 1),
         Replacement.delete(node.expression.end - 1, node.expression.end),
     ];
+
+    /* Handles trailing commas in array/object literals */
+    if (
+        isSpreadElement(node) &&
+        isArrayLiteralExpression(node.expression) &&
+        node.expression.elements.hasTrailingComma
+    ) {
+        replacements.push(Replacement.delete(node.expression.elements.end - 1, node.expression.elements.end));
+    } else if (
+        isSpreadAssignment(node) &&
+        isObjectLiteralExpression(node.expression) &&
+        node.expression.properties.hasTrailingComma
+    ) {
+        replacements.push(Replacement.delete(node.expression.properties.end - 1, node.expression.properties.end));
+    }
+
+    return replacements;
+}
+
+function removeUselessSpreadOfEmptyArray(node: ts.SpreadElement): Replacement {
```

Comment:
argh! forgot to save my comment before submitting review...

I'd like to avoid the code duplication:

```ts
const parent = node.parent!;
const list = parent.kind === ts.SyntaxKind.ArrayLiteralExpression ? parent.elements : parent.arguments!;
const index = list.indexOf(node);
return Replacement.delete(node.pos, index === list.length - 1 ? list.end : list[index + 1].pos);
```

Something similar could also be used above to handle trailing comma